babel/numbers.py

```diff
@@ -467,11 +467,15 @@ def _get_compact_format(number, compact_format, locale, fraction_digits=0):
             # equal to the number of 0's in the pattern minus 1
             number = number / (magnitude / (10 ** (pattern.count("0") - 1)))
             # round to the number of fraction digits requested
-            number = round(number, fraction_digits)
+            rounded = round(number, fraction_digits)
             # if the remaining number is singular, use the singular format
             plural_form = locale.plural_form(abs(number))
-            plural_form = plural_form if plural_form in compact_format else "other"
+            if plural_form not in compact_format:
+                plural_form = "other"
+            if number == 1 and "1" in compact_format:
```

Comment:
Maybe `elif`? 
```suggestion
            elif number == 1 and "1" in compact_format:
```

Reply:
If the number were `1`, the plural form could initially be "one", if "one" is not in `compact_format`, it will become "other", but if "1" is, it should still use "1".